tools/fastp/fastp.xml

```diff
@@ -0,0 +1,332 @@
+
+            #if str($cutting_by_quality_options.cut_window_size):
+                -W $cutting_by_quality_options.cut_window_size
+            #end if
+            #if str($cutting_by_quality_options.cut_mean_quality):
+                -M $cutting_by_quality_options.cut_mean_quality
+            #end if
+        #end if
+
+        ## quality filtering options
+        #if $quality_filtering_options.disable_quality_filtering:
+            $quality_filtering_options.disable_quality_filtering
+        #else:
+            #if str($quality_filtering_options.qualified_quality_phred):
+                -q $quality_filtering_options.qualified_quality_phred
+            #end if
+            #if str($quality_filtering_options.unqualified_percent_limit):
+                -u $quality_filtering_options.unqualified_percent_limit
+            #end if
+            #if str($quality_filtering_options.n_base_limit):
+                -n $quality_filtering_options.n_base_limit
+            #end if
+        #end if
+
+        ## length filtering options
+        #if $length_filtering_options.disable_length_filtering:
+            $length_filtering_options.disable_length_filtering
+        #else:
+            #if str($length_filtering_options.length_required):
+                -l $length_filtering_options.length_required
+            #end if
+        #end if
+
+        ## base correction by overlap analysis options
+        #if $base_correction_options.correction:
+            $base_correction_options.correction
+        #end if
+
+        ## UMI processing
+        #if $umi_processing.umi:
+            $umi_processing.umi
+            #if str($umi_processing.umi_loc):
+                --umi_loc '$umi_processing.umi_loc'
+            #end if
+            #if str($umi_processing.umi_len):
+                --umi_len $umi_processing.umi_len
+            #end if
+            #if str($umi_processing.umi_prefix):
+                --umi_prefix '$umi_processing.umi_prefix'
+            #end if
+        #end if
+
+        ## overrepresented sequence analysis
+        #if $overrepresented_sequence_analysis.overrepresentation_analysis:
+            $overrepresented_sequence_analysis.overrepresentation_analysis
+            #if str($overrepresented_sequence_analysis.overrepresentation_sampling):
+                -P $overrepresented_sequence_analysis.overrepresentation_sampling
+            #end if
+        #end if
+
+        ## thread options
+        --thread \${GALAXY_SLOTS:-1}
+        ]]>
+    </command>
+
+    <inputs>
+        <conditional name="single_paired">
+            <param name="single_paired_selector" type="select" label="Single-end or paired reads">
+                <option value="single" selected="true">Single-end</option>
+                <option value="paired">Paired</option>
+            </param>
+            <when value="single">
+                <param format="fastq" name="in1" type="data" label="Input 1" argument="-i" help="Input FastQ file #1."/>
```

Comment:
[ordering](https://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html#tool-descriptions) (scroll to the bottom): name, argument, type, format, label, help